pkg/custompluginmonitor/plugin/plugin.go

```diff
@@ -127,14 +147,62 @@ func (p *Plugin) run(rule cpmtypes.CustomRule) (exitStatus cpmtypes.Status, outp
 	defer cancel()
 
 	cmd := exec.CommandContext(ctx, rule.Path, rule.Args...)
-	stdout, err := cmd.Output()
+
+	stdoutPipe, err := cmd.StdoutPipe()
+	if err != nil {
+		glog.Errorf("Error creating stdout pipe for plugin %q: error - %v", rule.Path, err)
+		return cpmtypes.Unknown, "Error creating stdout pipe for plugin. Please check the error log"
+	}
+	stderrPipe, err := cmd.StderrPipe()
 	if err != nil {
+		glog.Errorf("Error creating stderr pipe for plugin %q: error - %v", rule.Path, err)
+		return cpmtypes.Unknown, "Error creating stderr pipe for plugin. Please check the error log"
+	}
+	if err := cmd.Start(); err != nil {
+		glog.Errorf("Error in starting plugin %q: error - %v", rule.Path, err)
+		return cpmtypes.Unknown, "Error in starting plugin. Please check the error log"
+	}
+
+	var wg sync.WaitGroup
+	var stdout []byte
+	var stderr []byte
+	var stdoutErr error
+	var stderrErr error
+
+	wg.Add(2)
+	go func() {
+		stdout, stdoutErr = readFromReader(stdoutPipe, maxCustomPluginBufferBytes)
+		wg.Done()
+	}()
+	go func() {
+		stderr, stderrErr = readFromReader(stderrPipe, maxCustomPluginBufferBytes)
+		wg.Done()
+	}()
+	wg.Wait()
```

Comment:
It seems that this may block forever.

In theory, it should not happen, because the command should be stopped after timeout, and the IO should be closed.

However, I'm not completely sure about that. Maybe we can call `wg.Wait` after `Wait` the command to stop?

Reply:
The documentation for stdoutpipe specifies: https://golang.org/src/os/exec/exec.go?s=17733:17782#L609
It is incorrect to call Wait before all reads from the pipe have completed.

Having the wg.Wait() after will call exec.Wait before we are done reading from pipes?